crates/vm/src/wasm/memory.rs

```diff
@@ -192,6 +199,26 @@ pub fn write_vp_inputs(
     })
 }
 
+/// Allocated inside the wasm instance using the injected function and return a
+/// ptr to it
+fn wasm_alloc(
+    instance: &wasmer::Instance,
+    store: &mut impl wasmer::AsStoreMut,
+    len: u64,
+) -> Result<u64> {
+    let alloc_fn = instance
+        .exports
+        .get_function(super::run::ALLOC_FN_NAME)
+        .unwrap()
+        .typed::<i32, i32>(&*store)
```

Comment:
it doesn't work with `u32` instead of `i32`? not that we'll have the need to allocate 4 GiB, but...